### PR TITLE
Do not discard cue types that can't be imported by SeratoCueImporter (lp1920087)

### PR DESCRIPTION
**Pull request by Holzhaus**
Launchpad issue: https://bugs.launchpad.net/mixxx/+bug/1920087

### CODE REVIEW

File: src/track/track.cpp [hunk 1]
```diff
@@ -1108,7 +1108,16 @@ bool Track::importPendingCueInfosWhileLocked() {
             m_record.getMetadata().getStreamInfo().getSignalInfo().getSampleRate());
     const auto trackId = m_record.getId();
     QList<CuePointer> cuePoints;
-    cuePoints.reserve(m_pCueInfoImporterPending->size());
+    cuePoints.reserve(m_pCueInfoImporterPending->size() + m_cuePoints.size());
+
+    // Avoid unnecessary data loss by not discarding/overwriting existing cues
```
Comment (uklotzde) [inline src/track/track.cpp:1113]: The comment is not easy to understand. How about "Preserve all existing cues with types that are not available for importing."

**Review by uklotzde — CHANGES_REQUESTED** [review]
If no (= 0) cues of a certain type are available for import we should not overwrite the corresponding existing cues, even if the cue type itself is supported. Just to be safe. Only preserving the "non-importable" cue types is not enough imho. To be discussed.

**Comment by Holzhaus** [comment]
> If no (= 0) cues of a certain type are available for import we should not overwrite the corresponding existing cues, even if the cue type itself is supported. Just to be safe. Only preserving the "non-importable" cue types is not enough imho. To be discussed.

I didn't implement this because I was worried if users would be confused by this behavior or the inconsistent state (e.g. new cues, but old loops if there are no loops in Serato's metadata). For users that don't want this, we should add a way to a way to reset only cues of a certain type then.

**Comment by uklotzde** [comment]
> > If no (= 0) cues of a certain type are available for import we should not overwrite the corresponding existing cues, even if the cue type itself is supported. Just to be safe. Only preserving the "non-importable" cue types is not enough imho. To be discussed.
> 
> I didn't implement this because I was worried if users would be confused by this behavior or the inconsistent state (e.g. new cues, but old loops if there are no loops in Serato's metadata). For users that don't want this, we should add a way to a way to reset only cues of a certain type then.

We generally do not delete any metadata on import if it is absent in file tags. As a user I consider different types of cues as different metadata that should not interfere with each other.

Deleting the obsolete cues manually might be tedious, but it is the safe option compared to silently deleting them. Adding some convenience functions for managing the different cue types independently would be a bonus. It is not strictly necessary.

**Comment by Holzhaus** [comment]
Changes added. If you think it's worth implementing a more efficient way that is not O(n*m) complexity, let me know. Since n <= 37, m <= 37 I did not bother because this probably makes the code harder to read.

**Comment by uklotzde** [comment]
> Changes added. If you think it's worth implementing a more efficient way that is not O(n*m) complexity, let me know. Since n <= 37, m <= 37 I did not bother because this probably makes the code harder to read.

That's still considered constant then ;)

**Review by uklotzde — APPROVED** [review]
Thanks! LGTM

**Comment by Holzhaus** [comment]
So it's O(1) :tada: and n is actuall <= 8 ;-)

**Pull request merged**